src/BuiltinExtensions/ComfyUIBackend/WorkflowGenerator.cs

```diff
@@ -326,18 +326,25 @@ public string CreateLoadImageNode(Image img, string param, bool resize, string n
         });
         string scaledImage = CreateNode("SwarmImageScaleForMP", new JObject()
         {
-            ["image"] = new JArray() { croppedImage, 0 },
-            ["width"] = UserInput.Get(T2IParamTypes.Width, 1024),
-            ["height"] = UserInput.GetImageHeight(),
-            ["can_shrink"] = false
+            ["image"] = new JArray() { croppedImage, 0 },            
+            ["width"] = GetRelevantModel(fromSegmentation)?.StandardHeight,
+            ["height"] = GetRelevantModel(fromSegmentation)?.StandardWidth,
+            ["can_shrink"] = true
         });
         string vaeEncoded = CreateVAEEncode(vae, [scaledImage, 0], null, true);
-        string masked = CreateNode("SetLatentNoiseMask", new JObject()
+        string latentNoiseMasked = CreateNode("SetLatentNoiseMask", new JObject()
         {
             ["samples"] = new JArray() { vaeEncoded, 0 },
             ["mask"] = new JArray() { croppedMask, 0 }
         });
-        return (boundsNode, croppedMask, masked);
+
+        return (boundsNode, croppedMask, skipLatentNoise ? vaeEncoded : latentNoiseMasked);
+    }
+
+    // Segmentation use refiner Model when enabled but inpainting does not
+    private T2IModel GetRelevantModel(bool fromSegmentation)
```

Comment:
This should probably be a sub-method inside of the method it's used by rather than a class method